packages/aws-cdk-lib/aws-ecs/lib/ec2/ec2-task-definition.ts

```diff
@@ -146,4 +147,23 @@ export class Ec2TaskDefinition extends TaskDefinition implements IEc2TaskDefinit
     // Validate the placement constraints
     Ec2TaskDefinition.validatePlacementConstraints(props.placementConstraints ?? []);
   }
+
+  /**
+   * Tasks running in AWSVPC networking mode requires an additional environment variable for the region to be sourced.
+   * This override adds in the additional environment variable as required
+   */
+  override addContainer(id: string, props: ContainerDefinitionOptions): ContainerDefinition {
+    if (this.networkMode === NetworkMode.AWS_VPC) {
+      return new ContainerDefinition(this, id, {
+        taskDefinition: this,
+        ...props,
+        environment: {
+          ...props.environment,
+          AWS_REGION: Stack.of(this).region,
+        },
+      });
+    }
+    // If network mode is not AWSVPC, then just add the container as normal
+    return new ContainerDefinition(this, id, { taskDefinition: this, ...props });
```

Comment:
```suggestion
    if (this.networkMode === NetworkMode.AWS_VPC) {
      return super.addContainer(id, {
        ...props,
        environment: {
          ...props.environment,
          AWS_REGION: Stack.of(this).region,
        },
      });
    }
    // If network mode is not AWSVPC, then just add the container as normal
    return super.addContainer(id, props);
```
Let's reuse the parent's class method